docs/technical-details/reference/resource-migration.md

```diff
@@ -0,0 +1,116 @@
+# Migration to non-instance Resource API
+
+The Resource API was heavily influenced/inspired by two major design ideas: the REST architectural design and the [Active Record pattern](https://en.wikipedia.org/wiki/Active_record_pattern) (made popular by Ruby on Rails, and heavily used as a pattern in many ORMs). The basic design goal of the Resource API is to integrate these concepts into a single construct that can directly map RESTful methods (specifically the "uniform interface" of HTTP) to an active record data model. However, while the active record pattern has been for _consumption_ of data, implementing methods for endpoint definitions and caching sources as a data _provider_ can be confusing and cumbersome to implement. The updated non-instance binding Resource API is designed to make it easier and more consistent to implement a data provider and interact with records across a table, while maintaining more explicit control over what data is loaded and when.
+
+The updated Resource API is enabled on a per-class basis, by setting static `loadAsInstance` property to `false`. When this property is set to `false`, this means that the Resource instances will not be bound to a specific record, and instead instances represent the whole table, capturing the context and current transactional state. Any records in the table can be loaded or modified from `this` instance. There are a number of implications and different behaviors from a Resource class with `static loadAsInstance = false`:
+* The `get` method (both static and instance) will directly return the record, a frozen enumerable object with direct properties, instead of a Resource instance.
+* When instance methods are called, there will not be any record preloaded beforehand. And the resource instance will not have properties mapped to a record.
+* All instance methods accept a `target`, an instance of `RequestTarget`, as the first argument, which identifies the target record or query.
+	* The `target` will have an `id` property identifying the target resource, along with target information. The `getId()` method will return `undefined`.
```

Comment:
I was confused when I read `getId()` will return `undefined`. I understand why, but will others be confused? Maybe it could be its own bullet item?

```suggestion
	* The `target` will have an `id` property identifying the target resource, along with target information.
	* The `getId()` method is not required and will return `undefined`.
```